src/main/java/org/traccar/protocol/Mavlink2ProtocolDecoder.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 AuroraRAS (Chipeng Li, chplee@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import java.net.SocketAddress;
+import java.text.DecimalFormat;
+import java.time.Instant;
+import java.util.Date;
+
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.DeviceSession;
+import org.traccar.Protocol;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.Position;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+
+public class Mavlink2ProtocolDecoder extends BaseProtocolDecoder {
+
+    public Mavlink2ProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+        ByteBuf packet = (ByteBuf) msg;
+        if (packet.readUnsignedByte() != 0xFD) {
+            return null;
+        }
+        int payloadLength = packet.readUnsignedByte();
+        packet.skipBytes(3); // incompat_flags, compat_flags, seq
+        int senderSystemId = packet.readUnsignedByte();
+        packet.skipBytes(1); // compid
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, Integer.toString(senderSystemId));
+        if (deviceSession == null) {
+            return null;
+        }
+        int messageId = (
+            packet.readByte() & 0xff) | ((packet.readByte() & 0xff) << 8) | ((packet.readByte() & 0x0f) << 16);
+        if (messageId == 33) {
+            ByteBuf message = packet.readBytes(payloadLength);
+            Position position = new Position(getProtocolName());
+            position.setDeviceId(deviceSession.getDeviceId());
+            position.setValid(true);
+            position.set("timeBootms", new DecimalFormat("#,###").format(message.readUnsignedIntLE()));
+            position.setTime(Date.from(Instant.now()));
+            position.setLatitude((double) message.readIntLE() / 10000000);
```

Comment:
```suggestion
            position.setLatitude(message.readIntLE() / 10000000.0);
```
And same for other fields as well. It's more readable this way.